dotnet/src/SemanticKernel/TemplateEngine/Blocks/NamedArgBlock.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.TemplateEngine.Blocks;
+
+internal sealed class NamedArgBlock : Block, ITextRendering
+{
+    internal override BlockTypes Type => BlockTypes.NamedArg;
+
+    internal string Name { get; } = string.Empty;
+
+    public string GetValue(ContextVariables? variables)
+    {
+        var valueIsValidValBlock = this._valBlock != null && this._valBlock.IsValid(out var errorMessage);
+        var valueIsValidVarBlock = this._argValueAsVarBlock != null && this._argValueAsVarBlock.IsValid(out var errorMessage2);
+        if (valueIsValidValBlock && this._valBlock != null)
```

Comment:
I included the null checks because I get this warning when I don't include them: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/nullable-warnings?f1url=%3FappId%3Droslyn%26k%3Dk(CS8602).

Reply:
You can use the `!` null-forgiving operator to suppress the warnings. It's safe because the null check has already been performed, and it's a little bit easier to read - there's no need to think about why the null check has been done two times.
<img width="791" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/2a8bd29b-dda4-4a9a-baf0-059407948af3">

Reply:
Forgot about that, good idea.